diff -aurN nginx-1.13.5-1.el7.ngx.src.orig/nginx.spec nginx-1.13.5-1.el7.ngx.src/nginx.spec
--- nginx-1.13.5-1.el7.ngx.src.orig/nginx.spec	2017-09-07 10:11:03.949952310 +0900
+++ nginx-1.13.5-1.el7.ngx.src/nginx.spec	2017-09-07 10:17:40.960937672 +0900
@@ -25,16 +25,27 @@
 BuildRequires: openssl-devel >= 1.0.1
 %endif
 
-%if 0%{?rhel} == 7
+%if (0%{?rhel} == 7 || 0%{?fedora} >= 18)
 %define _group System Environment/Daemons
 %define epoch 1
 Epoch: %{epoch}
 Requires(pre): shadow-utils
 Requires: systemd
-Requires: openssl >= 1.0.1
 BuildRequires: systemd
+
+%if (0%{?fedora} >= 26)
+Requires: compat-openssl10 >= 1.0.1
+BuildRequires: compat-openssl10-devel >= 1.0.1
+%else
+Requires: openssl >= 1.0.1
 BuildRequires: openssl-devel >= 1.0.1
-%define dist .el7
+%endif
+%endif
+
+%if 0%{?fedora} >= 18
+# removes -Werror in upstream build scripts.  -Werror conflicts with
+# -D_FORTIFY_SOURCE=2 causing warnings to turn into errors.
+Patch0: nginx-auto-cc-gcc.patch
 %endif
 
 %if 0%{?suse_version} == 1315
@@ -103,8 +114,14 @@
     -e 's|%%PROVIDES%%|nginx|g' < %{SOURCE2} > nginx.init
 sed -e 's|%%DEFAULTSTART%%||g' -e 's|%%DEFAULTSTOP%%|0 1 2 3 4 5 6|g' \
     -e 's|%%PROVIDES%%|nginx-debug|g' < %{SOURCE2} > nginx-debug.init
+%if 0%{?fedora} >= 18
+%patch0 -p0
+%endif
 
 %build
+%if 0%{?fedora} >= 26
+export CFLAGS="-Wno-error"
+%endif
 ./configure %{BASE_CONFIGURE_ARGS} \
     --with-cc-opt="%{WITH_CC_OPT}" \
     --with-ld-opt="%{WITH_LD_OPT}" \
