diff -aurN nginx-1.11.10-1.el7.ngx.src.orig/nginx.spec nginx-1.11.10-1.el7.ngx.src/nginx.spec
--- nginx-1.11.10-1.el7.ngx.src.orig/nginx.spec	2017-02-16 19:17:24.947789127 +0900
+++ nginx-1.11.10-1.el7.ngx.src/nginx.spec	2017-02-16 19:19:38.830784191 +0900
@@ -25,7 +25,7 @@
 BuildRequires: openssl-devel >= 1.0.1
 %endif
 
-%if 0%{?rhel} == 7
+%if (0%{?rhel} == 7 || 0%{?fedora} >= 18)
 %define _group System Environment/Daemons
 %define epoch 1
 Epoch: %{epoch}
@@ -36,6 +36,12 @@
 BuildRequires: openssl-devel >= 1.0.1
 %endif
 
+%if 0%{?fedora} >= 18
+# removes -Werror in upstream build scripts.  -Werror conflicts with
+# -D_FORTIFY_SOURCE=2 causing warnings to turn into errors.
+Patch0: nginx-auto-cc-gcc.patch
+%endif
+
 %if 0%{?suse_version} == 1315
 %define _group Productivity/Networking/Web/Servers
 %define nginx_loggroup trusted
@@ -102,6 +108,9 @@
     -e 's|%%PROVIDES%%|nginx|g' < %{SOURCE2} > nginx.init
 sed -e 's|%%DEFAULTSTART%%||g' -e 's|%%DEFAULTSTOP%%|0 1 2 3 4 5 6|g' \
     -e 's|%%PROVIDES%%|nginx-debug|g' < %{SOURCE2} > nginx-debug.init
+%if 0%{?fedora} >= 18
+%patch0 -p0
+%endif
 
 %build
 ./configure %{BASE_CONFIGURE_ARGS} \
