diff -aurN nginx-1.11.8-1.el7.ngx.src.orig/nginx.spec nginx-1.11.8-1.el7.ngx.src/nginx.spec > nginx-fedora.patch
--- nginx-1.11.8-1.el7.ngx.src.orig/nginx.spec	2016-12-28 10:29:57.315897713 +0900
+++ nginx-1.11.8-1.el7.ngx.src/nginx.spec	2016-12-28 10:31:41.812893860 +0900
@@ -25,7 +25,7 @@
 BuildRequires: openssl-devel >= 1.0.1
 %endif
 
-%if 0%{?rhel} == 7
+%if (0%{?rhel} == 7 || 0%{?fedora} >= 18)
 %define _group System Environment/Daemons
 %define epoch 1
 Epoch: %{epoch}
